types/wlr_text_input_v3.c

```diff
@@ -61,6 +61,9 @@ void wlr_text_input_v3_send_done(struct wlr_text_input_v3 *text_input) {
 
 static void wlr_text_input_destroy(struct wlr_text_input_v3 *text_input) {
 	wlr_signal_emit_safe(&text_input->events.destroy, text_input);
+	if (text_input->focused_surface) {
```

Comment:
Instead of adding conditions like this, you can always call `wl_list_remove`, and call `wl_list_init` when the surface is removed.